src/process_utility.c

```diff
@@ -365,8 +365,22 @@ process_copy(ProcessUtilityArgs *args)
 	Cache *hcache;
 	Oid relid;
 
+	/* We only copy for COPY FROM. Since hypertables do not contain any data,
+	 * we skip the copy for COPY TO, but print an informative message. */
 	if (!stmt->is_from || NULL == stmt->relation)
+	{
+		if (stmt->relation)
+			ereport(NOTICE,
+					(errmsg("hypertables has no data in main table, no data will be copied"),
```

Comment:
```suggestion
					(errmsg("hypertables do not support COPY TO"),
```

Reply:
IMHO saying `hypertables do not support COPY TO` is misleading. Such message requires error, not notice.

Reply:
But that would break `pg_dump`, which we already know.

Reply:
It is not strictly true that it does not support `COPY TO`. Using a sub-query works fine.

Reply:
> Using a sub-query works fine.

Subquery is not a hypertable. It selects from it.

Reply:
Users are unlikely to be aware of the distinction.